api/src/main/java/org/openmrs/Obs.java

```diff
@@ -550,11 +550,9 @@ public void removeGroupMember(Obs member) {
 				parentObs = parentObs.getObsGroup();
 			}
 		} else if (this.getObsGroup() != null) {
-			for (Obs obsSibling : this.getObsGroup().getGroupMembers()) {
-				if (!obsSibling.isObsGrouping()) {
-					ret.add(obsSibling);
-				}
-			}
+                    this.getObsGroup().getGroupMembers().stream().filter((obsSibling) -> (!obsSibling.isObsGrouping())).forEachOrdered((obsSibling) -> {
```

Comment:
Can you let the stream api build the collection instead of you manually adding the elements one by one?

Reply:
Yes you can remove the nesting with streams and at the same time not have to manually add items to the collection.